lib/cog/models/types/version_triple.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Cog.Models.Types.VersionTriple do
+
+
+  @version_separator "."
+
+
+  # type should output the name of the db type
+  # cast should receive any type and output your custom Ecto type
+  # load should receive the db type and output your custom Ecto type
+  # dump should receive your custom Ecto type and output the db type
+
+  @behaviour Ecto.Type
+  def type,
+    do: :array
+
+  def cast(version_string) when is_binary(version_string) do
+    case version_string
+    |> String.split(@version_separator)
+    |> Enum.map(&Integer.parse/1) do
+      [{_major, ""}, {_minor, ""}, {_patch, ""}] ->
+        #{:ok, [major, minor, patch]}
+        {:ok, version_string}
+      _ ->
+        :error
+    end
+  end
```

Comment:
An earlier implementation used `Version` but it wasn't working for some reason... I need to revisit that, though.